### PR TITLE
First Structure of the bootstrap project.

### PR DESCRIPTION
**Pull request by SigmaSam**
Build a recreation of the Newsweeks website using mostly using bootstrap classes.
Features
-Responsiveness for all screens.

### CODE REVIEW

**Comment by SigmaSam** [comment]
Build a recreation of the Newsweeks website using mostly using bootstrap classes.
Features
-navigation bars hiding or showing according to the size of the user screen.
-Responsiveness for all screen sizes.
-Full use of bootstrap classes.

**Comment by elmejdki** [comment]
## Pull Request under review by _@Zakariae El Mejdki_

File: README.md [hunk 1]
```diff
@@ -3,7 +3,9 @@ This project requests students to build a replica of the news site Newsweek.com
 
  > I replicated the newssite Newsweek.com including their responsiveness for smaller screens.
  ![screenshot](Images/screenshot1.png)
+
  ![screenshot](Images/screenshot3.png)
+
  ![screenshot](Images/screenshot2.png)
```
Comment (elmejdki) [inline README.md]: Fix the links in the images it's not working maybe you are not using an upper case 'i' letter in images.

**Review by elmejdki — CHANGES_REQUESTED** [review]
# Good job so far @SigmaSam :clap: :clap: :clap:, your version looks really good :tophat:, but you need to do some changes in order to make your project approved.
### status: require changes.

- Fix the markdown to use a link you need to remove the space between ] and (.
image

- **[OPTIONAL]** add your LinkedIn and Twitter profile links also in your readme.

- Please add the Current date on the left top corner of your navbar as it is on the original page.
image

- your navbar menu breaks down on small screen size please fix it.
### Desktop view:
#### Your version:
:point_down:     :point_down:      :point_down:
image

#### Original page:
:point_down:       :point_down:       :point_down: 
image

### Mobile view:
#### your version:
:point_down:     :point_down:      :point_down:
image

##### Original page:
image

- you have a section the disappears in the top story section on tablet view:
image

- Please fix the width of these two images on mobile version somehow they are not appearing as they should:
image

- Please fix this section on mobile view make it similar to the original one:
##### your version:
image


##### original page:
image

- Please fix the newsletter and subscribe section to look similar to the original, on mobile and desktop view:
#### Desktop view:
##### Your version:
image

##### Original page:
image

#### Mobile view:
##### Your version:
image

##### Original page:
image

### please fix all the required changes before you request another code review.
_If you have any questions don't hesitate to contact me on slack_

**Comment by SigmaSam** [comment]
Fixed the reviewd issues.

**Review by HADDADSOHAIB — CHANGES_REQUESTED** [review]
### Status: Change Required :recycle: 

### Hello @SigmaSam   :hand: 

### Good job your project is getting better and better. :+1: 

### Unfortunately, there are still some changes to get the project approved:
- Please fix the conflict:
Screenshot from 2020-06-15 12-11-41
- Please fix the images on README file:
Screenshot from 2020-06-15 12-20-31
- I think this part of the mobile view still needs some improvement:
your project:
Screenshot from 2020-06-15 12-26-57
Original:
Screenshot from 2020-06-15 12-27-31
Think of reducing the margins and increasing the size of small images a little bit.
- The newsletter that you implemented is different then the original:
desktop, your project:
Screenshot from 2020-06-15 12-32-06
desktop, original:
Screenshot from 2020-06-15 12-32-31
mobile, your project:
Screenshot from 2020-06-15 12-33-18
mobile, original:
Screenshot from 2020-06-15 12-33-55
### Once you are done with the changes, ask for another code review.

**Comment by SigmaSam** [comment]
-Fixed conflict in the README.MD file.
-Fixed image size on smaller screens.
-Add image to newsletter in desktop screens.

**Comment by SigmaSam** [comment]
-Fixed conflict in the README.MD file.
-Fixed image size on smaller screens.
-Add image to newsletter in desktop screens.

**Review by HADDADSOHAIB — CHANGES_REQUESTED** [review]
### Status: Change Required :recycle: 

### Hello @SigmaSam   :hand: 

### Thanks for fixing the readme file, but the other issues are not fixed. :+1: 
- As you can see here this part need a lot of improvement, focus of the margins or paddings, they are the ones that cause this, also the size of the font on mobile view should be smaller then the desktop view:
your project:
Screenshot from 2020-06-15 14-32-02

Original:
Screenshot from 2020-06-15 12-27-31
Also, the horizontal scrolling should not be there, I am sorry I did not signal that before, all you need to do is to add this to css file:
`body {
    overflow-x: hidden;
}`
- The newsletter still need improvements:
mobile, your project:
Screenshot from 2020-06-15 12-33-18
mobile, original:
Screenshot from 2020-06-15 12-33-55
### Once you are done with the changes, ask for another code review.

**Comment by SigmaSam** [comment]
-Set a fixed image size.
-Set the subscribe section as a column in mobile resolutions.

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @SigmaSam, great work fixing some of the comments made by the previous TSE :+1: However, there were some comments that were not addressed. I would like to highlight them so they get fixed. Kindly find comments below:
- ```Main navigation``` our main navigation needs to be fixed
mainN
- Kindly fix the issues mentioned by the previous TSE's about the Subscribe and Newsletter section. It looks like the same from the previous TSE's comments.

Kindly make these changes and submit another code review :+1: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Comment by SigmaSam** [comment]
-Fixed the size of the font of the second navbar.

I honestly can't see the issue with the subscribe/newsletter here is a screen capture of how it looks in my screen, maybe is some issue with the browsers so I really don't know where to tackle it.

image

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @SigmaSam, great work, and patience :+1: The following is a screenshot of the clone, source, and requirements:
real1

real
real2
 
Per this requirement, we are advised to work to fix it. A quick fix will be to set a navContainer width: 100%; display: grid; grid-template-columns: repeat(no of list items, 1fr) or contact me if you may need assistance with this

Kindly submit a code review when done :+1: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Comment by SigmaSam** [comment]
-Expanded the witdth of the second navigation bar as asked.

**Comment by andresporras3423** [comment]
### **Pull Request Under Review By Oscar Russi (TSE)**

File: styles.css [hunk 2]
```diff
@@ -0,0 +1,578 @@
...
+.letter__center {
+  text-align: center;
+}
+
+.letter__title {
+  font-weight: bold;
+}
+
+.let__button {
+  background-color: #f72210 !important;
```
Comment (andresporras3423) [inline styles.css]: Don't overuse important

File: styles.css [hunk 3]
```diff
@@ -0,0 +1,578 @@
...
+
+.foot__link a {
+  color: #666;
+}
+
+#see {
+  display: none;
+}
+
+#see_two a {
```
Comment (andresporras3423) [inline styles.css]: Don't overuse ID Selectors

File: styles.css [hunk 4]
```diff
@@ -0,0 +1,578 @@
...
+    fill: white;
+  }
+}
+
+@media (max-width: 768px) {
+  #hero {
+    display: none;
+  }
+
+  #more {
```
Comment (andresporras3423) [inline styles.css]: Don't overuse ID selectors

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**@SigmaSam** 
Good job with your page, it really looks like the original :muscle:  
Your page pass most of the requirements  :clap: :clap: :clap:  
There are just a few details to improve in your CSS code before you can go ahead:
- [x] Don't use important
- [x] Don't overuse ID Selectors

Please request another review once you are ready! :+1: 

Happy coding :relaxed:

Oscar Andrés Russi Porras
(for any question you can contact me in slack as @oscarporras02)

**Comment by SigmaSam** [comment]
-Removed ID's.
-Removed !important.

**Review by tirthajyoti-ghosh — CHANGES_REQUESTED** [review]
# Status: Changes required 🚨

## Great job implementing the changes 👏🙌!

### But we still need to work on some issues.

- The mobile version has extra space on the right side. The content should be full width in mobile version.

image

- In `the debate` section, the images are too large, in all screen sizes.

image

image

image

- This part of the navbar needs fixing.

Annotation 2020-06-17 004426


Apply these changes and request another code review.

Happy coding 👨‍💻

**Comment by SigmaSam** [comment]
-Reduced the size of the debate container
-Fixed the size of the corner decoration of the second navigation bar.

**Comment by andresporras3423** [comment]
### **Pull Request Under Review By Oscar Russi (TSE)**

File: styles.css [hunk 5]
```diff
@@ -0,0 +1,586 @@
...
+.debater__two {
+  border-bottom: 0.4rem solid #1c87fb;
+  padding-bottom: 1rem;
+}
+
+.debate__art {
+  color: black;
+  font-size: 1rem;
+  font-weight: bold;
+}
```
Comment (andresporras3423) [inline styles.css]: This selector has the same styles as the selector on line 249. Group selectors to minimize code in your CSS file.

File: styles.css [hunk 6]
```diff
@@ -0,0 +1,586 @@
...
+
+.debater__one img {
+  object-fit: scale-down;
+  max-height: 90px;
+}
+
+.debater__two img {
+  object-fit: scale-down;
+  max-height: 90px;
+}
```
Comment (andresporras3423) [inline styles.css]: Don't repeat yourself. Group selectors to minimize code in your CSS file.

File: styles.css [hunk 7]
```diff
@@ -0,0 +1,586 @@
...
+.debate__art {
+  color: black;
+  font-size: 1rem;
+  font-weight: bold;
+}
+
+.debate__art:hover {
+  text-decoration: none;
+  color: red;
+}
```
Comment (andresporras3423) [inline styles.css]: This selector has the same styles as the selector on line 255. Group selectors to minimize code in your CSS file.

File: styles.css [hunk 8]
```diff
@@ -0,0 +1,586 @@
...
+
+@media (max-width: 768px) {
+  .hero {
+    display: none;
+  }
+
+  .more {
+    display: none;
+  }
```
Comment (andresporras3423) [inline styles.css]: Don't repeat yourself. Group selectors to minimize code in your CSS file.

File: styles.css [hunk 9]
```diff
@@ -0,0 +1,586 @@
...
+    margin-right: -8rem;
+  }
+
+  .priv {
+    margin-bottom: -5rem;
+  }
+
+  .in_story {
+    margin-top: -5rem;
+  }
```
Comment (andresporras3423) [inline styles.css]: Group selectors to minimize code in your CSS file.

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello @SigmaSam** 
Good job so far  :muscle:  
I can see you have made a lot of improvements on your page :clap: :clap: :clap:  
You have to improve some parts:
- [x] Don't repeat yourself, in order to minimize code you can group selectors with the same style definition (in your CSS files).
- [x]  Check the navbar for missing details, compare the original:



With your clone:



Your page shows the hamburger icon even in the large screen, the hamburger icon should only appear in medium or smaller screen size version of the page. The search icon should be inside the input, not outside. Center the links of the navbar.

Now, let's check the footer, the original:



And your footer:



Use all the width of the page for the link in your footer. The text at the right side of the page icon should keep in one line. The newsletter title should be above the image, not on the right side.

Check the "in the magazine" section.  It looks like this in your page:



Please request another review once you are ready! :+1: 

Happy coding :relaxed:

Oscar Andrés Russi Porras

**Comment by SigmaSam** [comment]
- Cleaned the code from unnecessary classes.
- Inserted glass icon in the navigation search box.
- Fixed word warp in some parts of the general document.
- Widened the footer for bigger resolutions.

**Review by peterrobert — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED :recycle: :stop_sign: :mechanic: 
**HELLO** @SigmaSam :wave: 
**GREAT PROJECT** You have made the changes required by the previous TSE :+1: 
However, before we approve this project you need to do the following changes:

- [ ] There need to be some space on the left and right on your webpage

**original**
right_left_original

- [ ] Your menu items are being hidden in the medium view. Kindly improve on that

**Yours**
menu_items_being_hidden

- [ ] At mobile view your webpage is losing responsiveness on the menu
**yours**
menu_looses_repo_mobile

**original**
menu-looses_repo_mobile_original

**kindly improve on this before you send another code review** :smile: 

If you have any question about the review kindly contact me via slack on @peterrobert

### HAPPY CODING  :muscle:

**Comment by SigmaSam** [comment]
- Add the horizontal padding.
- Add no wrapping setting to the small screen navigation bar.

**Review by peterrobert — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED :recycle: 
HEY @SigmaSam :wave:  
Great work implementing the requested changes :clap:  However there are a few more things you need to check.

- [ ] Kindly, You have not yet improved on your menu bar. You need to make sure it looks like the clone.
**original**
search_bar_original
**yours**
search_bar

- [ ] Also you need to make sure the menu bar is responsive. That is it moves with the container. This is because when you reduce the screen size the search bar is being hidden

**yours**
search_disapper

- [ ] Great job applying the changes that were required on the mobile menu. However, the hamburger icon is moving off the container  
**hint: make the navigation move with the container. This will increase responsiveness**

**yours**
button_off_container

**Kindly make those changes before you ask for another review** :smile: 

if you have any question concerning the review kindly contact me via slack @peterrobert 

### HAPPY CODING :muscle:

**Comment by SigmaSam** [comment]
- Resized second navigation bar.
- Set the navigation bar in smaller resolutions as a wraping one.

**Review by HADDADSOHAIB — APPROVED** [review]
### Status: Project Approved :+1:  

### Hello @SigmaSam   :hand: 

### Good job, you did great job with the project :+1: 

### You can merge this branch into master.

### There are some additional requirements that are **optional**:
- [Optional] You should maintain a professional github repository in all of your projects, that includes removing unnecessary files.
Screenshot from 2020-06-17 11-47-48
Screenshot from 2020-06-17 11-47-40
Screenshot from 2020-06-17 11-47-34

**Pull request merged**